File: 0217-contains-duplicate/0217-contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        hashSet = set()
        
        
        for i in nums:
            if i not in hashSet:
                hashSet.add(i)
            else:
                return True
            
        return False
    

#Set
#Time Complexity: O(n)
#Space Complexity: O(n)

File: 0217-contains-duplicate/0217-contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        // set = hashset<int>;
        
       Set <Integer> set = new HashSet<Integer>();
        
        // HashMap<Integer, Integer> map = new HashMap<>();
        
        
        for(int i = 0; i< nums.length; i++){
            if(!set.contains(nums[i])){
                set.add(nums[i]);                
            }
            else{
                return true;
            }
        }        
        return false;
        
    }
}


//HashMap
//Time Complexity: O(n)
//Space Complexity: O(n)